ide-support/revliburl.livecodescript

```diff
@@ -5290,6 +5309,345 @@ private function _proxyfromHTTPProxy pURL, pProxy
+         end if
+         put "secure" is among the keys of tCookieA into tCookieA["secure"]
+         
+         ## Store each unique name=value pair on a separate line.
+         ## This allows us to easily update an existing cookie later on
+         repeat for each key tKey in tCookieA
+            if tKey is among the items of "expires,domain,path,secure" then next repeat
+            
+            ## is there already an entry for this cookie name?
+            put lineOffset(tKey & "=", laCookies[ tCookieA["domain"] ]) into tCookieLineNo
+            if tCookieLineNo is 0 then
+               put the number of lines of laCookies[ tCookieA["domain"] ] + 1 into tCookieLineNo
+            else
+               ## A cookie matches on name/domain/path. Make sure path matches.
+               put line tCookieLineNo of laCookies[ tCookieA["domain"] ] into tCheck
+               split tCheck by ";" and "="
+               if tCheck["path"] is not tCookieA["path"] then
+                  put the number of lines of laCookies[ tCookieA["domain"] ] + 1 into tCookieLineNo
+               end if
+            end if
+            
+            put tKey & "=" & tCookieA[tKey] & ";" & \
+                  "expires=" & tCookieA["expires"] & ";" & \
+                  "domain=" & tCookieA["domain"] & ";" & \
+                  "path=" & tCookieA["path"] & ";" & \
+                  "secure=" & tCookieA["secure"] & ";" into line tCookieLineNo of laCookies[ tCookieA["domain"] ]
+         end repeat
+      else
+         exit repeat
+      end if
+   end repeat
+   
+   -- ulLogIt laCookies[ tCookieA["domain"] ]
+   
+   return empty
+end _extractCookies
+
+
+private function _extractPath pURL
+   local tCharNo
+   local tPath
+
+   put "/" into tPath
+   put offset("://", pURL) into tCharNo
+   if tCharNo > 0 then
+      put char tCharNo + 3 to -1 of pURL into pURL
+      put offset("/", pURL) into tCharNo
+      if tCharNo > 0 then
+         put char tCharNo to -1 of pURL into tPath
+         put offset("?", tPath) into tCharNo
+         if tCharNo > 0 then
+            delete char tCharNo to -1 of tPath
+         end if
+      end if
+  end if
+
+  return tPath
+end _extractPath
+
+
+function libURLInternetTimeStampInGMT
+   return _ConvertInternetTimeStampToGMT(the internet date)
+end libURLInternetTimeStampInGMT
+
+
+function libURLTimeStampInGMT
+   return _ConvertInternetDateToGMTTimestamp(the internet date)
+end libURLTimeStampInGMT
+
+
+function libURLInternetDateToGMTTimeStamp pInternetDate
+   return _ConvertInternetDateToGMTTimestamp(pInternetDate)
+end libURLInternetDateToGMTTimeStamp
+
+
+function libURLGMTTimeStampToInternetDate pTimestamp
+   if char 1 of pTimestamp is an integer then
+      ## gmt timestamp
+      return _ConvertGMTTimestampToInternetDate(pTimestamp)
+   else
+      ## gmt internet timestamp
+      return _ConvertGMTInternetTimestampToInternetDate(pTimestamp)
+   end if
+end libURLGMTTimeStampToInternetDate
```

Comment:
It would be great to have some tests for the time-related functions.  Did you figure out how to run the testsuite locally yet?